lib/eslint.js

```diff
@@ -958,7 +958,7 @@ module.exports = (function() {
             // if current node is function declaration, add it to the list
             var current = controller.current();
             if (["FunctionDeclaration", "FunctionExpression",
-                    "ArrowFunctionExpression"].indexOf(current.type) >= 0) {
+                    "ArrowFunctionExpression", "SwitchStatement"].indexOf(current.type) >= 0) {
```

Comment:
What about loops? Sorry, I'm totally not up to date on es6 stuff, so it might be a meaningless comment.